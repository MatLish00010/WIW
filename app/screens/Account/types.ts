import { NativeStackScreenProps } from '@react-navigation/native-stack';
import { AccountStackParamList } from '@myapp/navigation/AccountStack/types';

export type AccountScreenProps = NativeStackScreenProps<AccountStackParamList>;
