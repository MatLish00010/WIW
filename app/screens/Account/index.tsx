import { FC } from 'react';
import { ActivityIndicator } from 'react-native';

import { TabScreenProps } from '@myapp/navigation/RootBottomTabs/types';
import useLogout from '@myapp/query/useLogout';

import useUser from '@myapp/hooks/useUser';
import { palette, spacing } from '@myapp/ui/Theme';
import { View, ViewContainer, Text, ButtonText, Button } from '@myapp/ui';

const Account: FC<TabScreenProps<'Account'>> = ({ navigation }) => {
  const { user } = useUser();
  const { isLoading, mutateAsync: logoutHandler } = useLogout();

  const handleLogout = async () => {
    await logoutHandler();
  };

  if (isLoading) {
    return (
      <View testID="loading" justifyContent="center" alignItems="center">
        <ActivityIndicator size="large" color={palette.blue} />
      </View>
    );
  }

  if (!user) {
    return (
      <View testID={'account-screen-user-is-not-registered'} alignItems="center" justifyContent="center">
        <Text mb={spacing.XL}>You must be logged in to access your account.</Text>
        <Button testID="signIn-btn" onPress={() => navigation.navigate('AuthStackNavigator', { screen: 'SignIn' })}>
          <ButtonText>Sign In</ButtonText>
        </Button>
      </View>
    );
  }

  return (
    <ViewContainer testID={'account-screen-user-is-registered'}>
      <Text mb={spacing.M}>Account</Text>
      <Text mb={spacing.M}>Email: {user.email}</Text>
      <Button testID="logout-btn" variant={'Error'} mt={spacing.XXL} onPress={handleLogout}>
        <ButtonText>Logout</ButtonText>
      </Button>
    </ViewContainer>
  );
};

export default Account;
