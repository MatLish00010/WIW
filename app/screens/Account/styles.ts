import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    marginBottom: 30,
  },
  loginButton: {
    width: 150,
    height: 150,
    borderRadius: 75,
    justifyContent: 'center',
  },
});
