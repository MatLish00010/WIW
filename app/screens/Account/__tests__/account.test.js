import Account from '@myapp/screens/Account';
import renderWithAllWrappers from '@myapp/utils/tests/renderWithAllWrappers';
import { render, cleanup } from '@testing-library/react-native';

describe('Account Page Test', () => {
  test('Page is correct', () => {
    render(renderWithAllWrappers(<Account />));
  });
  afterEach(cleanup);
});
