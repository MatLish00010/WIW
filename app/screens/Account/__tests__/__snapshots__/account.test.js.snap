// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Screen Test Screen for user not registered 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
  testID="account-screen-user-is-not-registered"
>
  <Text
    mb="32px"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 16,
          "marginBottom": 32,
        },
      ]
    }
  >
    You must be logged in to access your account.
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2f95dc",
        "borderRadius": 8,
        "display": "flex",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 16,
      }
    }
    testID="signIn-btn"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 16,
          },
        ]
      }
    >
      Sign In
    </Text>
  </View>
</View>
`;

exports[`Account Screen Test Screen for user registered 1`] = `
<View
  style={
    Array [
      Object {
        "height": "100%",
        "paddingBottom": 32,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 40,
      },
    ]
  }
  testID="account-screen-user-is-registered"
>
  <Text
    mb="16px"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 16,
          "marginBottom": 16,
        },
      ]
    }
  >
    Account
  </Text>
  <Text
    mb="16px"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 16,
          "marginBottom": 16,
        },
      ]
    }
  >
    Email: 
    xxx@xxx.xxx
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#eb0a3c",
        "borderRadius": 8,
        "display": "flex",
        "justifyContent": "center",
        "marginTop": 40,
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 16,
      }
    }
    testID="logout-btn"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 16,
          },
        ]
      }
    >
      Logout
    </Text>
  </View>
</View>
`;

exports[`Account Screen Test Screen is loading 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
  testID="loading"
>
  <ActivityIndicator
    color="#2f95dc"
    size="large"
  />
</View>
`;

exports[`Account screen test Screen for user not registered 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
  testID="account-screen-user-is-not-registered"
>
  <Text
    mb="32px"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 16,
          "marginBottom": 32,
        },
      ]
    }
  >
    You must be logged in to access your account.
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2f95dc",
        "borderRadius": 8,
        "display": "flex",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 16,
      }
    }
    testID="signIn-btn"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 16,
          },
        ]
      }
    >
      Sign In
    </Text>
  </View>
</View>
`;

exports[`Account screen test Screen for user registered 1`] = `
<View
  style={
    Array [
      Object {
        "height": "100%",
        "paddingBottom": 32,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 40,
      },
    ]
  }
  testID="account-screen-user-is-registered"
>
  <Text
    mb="16px"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 16,
          "marginBottom": 16,
        },
      ]
    }
  >
    Account
  </Text>
  <Text
    mb="16px"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 16,
          "marginBottom": 16,
        },
      ]
    }
  >
    Email: 
    xxx@xxx.xxx
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#eb0a3c",
        "borderRadius": 8,
        "display": "flex",
        "justifyContent": "center",
        "marginTop": 40,
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 16,
      }
    }
    testID="logout-btn"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 16,
          },
        ]
      }
    >
      Logout
    </Text>
  </View>
</View>
`;

exports[`Account screen test Screen is loading 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
  testID="loading"
>
  <ActivityIndicator
    color="#2f95dc"
    size="large"
  />
</View>
`;
